### ISSUE TITLE
setup.py

### ISSUE THREAD

**Issue by simevo** [techtonik/python-patch#51]
Hi, while [packaging this module for debian](https://bugs.debian.org/845482), we noticed that the `setup.py` file is missing.

This file is present in the ZIP from pip, but it would be nice for us if it were in the tarball from github as well.

Thanks, Paolo

**Comment by techtonik** [comment]
@simevo thanks for packaging it for Debian. I deliberately avoided `setup.py`, because I think it is anti-pattern to embed package metadata into executable script. It is hard to parse, hard to update with automated tools. Luckily there are new tools like `flit` that are following the same idea, so I need to review them to see if my custom `pypack` can already use package `source` without `setup.py` at all.

**Comment by techtonik** [comment]
Found https://www.python.org/dev/peps/pep-0517/ for alternative format of source packages without `setup.py`. Can Debian Python Policy adopt it?

I can move to this format as well, but as you already found out I don't like micromanaging metadata about simple Python modules when all necessary info is already self-contained. Maybe you could reuse https://github.com/techtonik/pypack for Debian packaging as well.
